### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by derpachu** [derpachu/MI-449-SS17-740-js-firebase-intro-T4LtxO#1]
## Make an anonymous messaging app that anyone can moderate

@KatieMFritz Can you take a look at this? It's [hosted here](https://tangjos1-kgucykegmdzxwt4ltxo.surge.sh/) and meets the following criteria:

- [x] Use [this code](https://gist.github.com/egillespie/942cc3aed19977590b431c10333736c2) as a foundation for your project
- [x] Your own Firebase project is used to store all of the site's messages
- [x] A message and its creation time are added to the database when Enter is pressed or Woof is clicked
- [x] Messages are stored in the database according to the structure in `model.json`
- [x] Anyone can edit any message by clicking the pencil icon, typing a new message, and pressing Enter
- [x] Anyone can delete any message by clicking the red X icon
- [x] When a message is added, edited, or deleted, _Recent woofs_ automatically updates for everyone viewing the page
- [x] Use the `TODO` comments in `woofer-db.js` as your guide for placing code

I wasn't sure if the time was also supposed to be updated when the pencil thing is clicked and changed

**Comment by egillespie** [comment]
Hi @derpachu, it looks like you're off to a really good start. There are a few recommendations I have to help clean up your code and simplify the organization of your data:

### Re: Updating `created_at` when the pencil icon is clicked

Can you remove this code? I'd like to see how you go about updating a single field on a record. 😄 

### Remove `{` and `}` that surround your woof in `createWoofInDatabase `

When you call `push({ woof })` you probably noticed that a new record is added to your database and it looks like this:

```js
woofs: {
  '98asfd97sadf9': {
    woof: {
      text: 'yolo',
      created_at: 987198174124
    }
  }
}
```

If you remove the `{` and `}` that surround `woof`, you'll get a much more concise structure that will be easier to work with:

```js
woofs: {
  '98asfd97sadf9': {
    text: 'yolo',
    created_at: 987198174124
  }
}
```

That means you can access the record with `firebase.database().ref('woofs').child(woofKey)` instead of with `firebase.database().ref('woofs/' + woofKey + '/woof')` like you're doing now.

Once you make this change, you will also be able to revert to the original `woofer-ui.js` file that we provided. We didn't intend for you to change that file.

---
After you’ve made your changes and pushed them to GitHub and your hosted site, give it a once-over to make sure it looks right, then comment back here and I’ll take another look.

Thanks! 🍔

**Comment by derpachu** [comment]
made the changes. Did you want me to reference the file by using the .child? Also I don't think I made any changes to the woofer-ui.js file

**Comment by egillespie** [comment]
> Did you want me to reference the file by using the .child?

Nope! From my experience this is purely a "coder's preference" and you're welcome to use whichever method you like best. 😃 

> Also I don't think I made any changes to the woofer-ui.js file

You're right. I dunno what I was smoking. 🚭 I think the extra layer in the database structure threw me off.

@derpachu, this is outstanding. Good work! :shipit:
